src/prover_calculi/superposition.ml

```diff
@@ -1544,6 +2616,31 @@ module Make(Env : Env.S) : S with module Env = Env = struct
     Util.exit_prof prof_clc;
     res
 
+  let inject_destruct lit =
```

Comment:
sounds like this could be in its own theory; after all it's close to a subset of what the Datatypes theory already does?

Reply:
I will remove this
This does not work nicely

How can we turn on by default what Datatypes does?

Reply:
I forgot, honestly. Look in the appropriate module :p 
(my point is just, these could probably be shared rules)

Reply:
removed